docs/03_stratum1_proxies.md

```diff
@@ -55,54 +88,125 @@ cache deny !CVMFSAPI
+
+#### Create and add Geo API key (optional)
+
+We first add our Geo API key to the CernVM-FS server settings, by creating it via
+and then running these commands:
+
+```bash
 echo 'CVMFS_GEO_LICENSE_KEY=YOUR_KEY' | sudo tee -a /etc/cvmfs/server.local
 sudo chmod 600 /etc/cvmfs/server.local
 ```
 
-We also need to have the public keys of all repositories we want to mirror to be available on our Stratum 1. This can be done by copying all the corresponding `.pub` files from `/etc/cvmfs/keys` on your Stratum 0 server to
-`/etc/cvmfs/keys/organization.tld/` (note the extra level!) on your Stratum 1 server.
+*Replace `YOUR_KEY` with your Geo API license key; see [https://www.maxmind.com/en/accounts/YOUR_ACCOUNT_ID/license-key](https://www.maxmind.com/en/accounts/YOUR_ACCOUNT_ID/license-key)!*
+
+Note that this is not strictly required for the sake of this tutorial, but it's highly recommended.
+
+#### Add repository master public key
+
+We also need to have the public master key of each repository we want to mirror to be available on our Stratum 1.
+
+This can be done by copying the `.pub` file(s) from `/etc/cvmfs/keys` on the Stratum 0 server to
+`/etc/cvmfs/keys/organization.tld/` (note the extra level!) on the Stratum 1 server, just like we did on the client.
 
-Now we make the replica by giving the URL to the repository on the Stratum 0 server (which is always like `http://host:port/cvmfs/repository`) and the path to the corresponding public key:
+#### Create replica
+
+Now we make the replica by giving the URL to the repository on the Stratum 0 server
+(which is always like `http://host:port/cvmfs/repository`, with the `:port` part optional)
+and the path to the corresponding public master key:
+
+```bash
+sudo cvmfs_server add-replica -o $USER http://<STRATUM0_IP>/cvmfs/repo.organization.tld /etc/cvmfs/keys/organization.tld/
 ```
-sudo cvmfs_server add-replica -o $USER http://YOUR_STRATUM0/cvmfs/repo.organization.tld /etc/cvmfs/keys/organization.tld/repo.organization.tld
+
+***Replace the ``<STRATUM0_IP>`` part with the IP address of your Stratum 0 server,
+and adjust for the name and domain of your CernVM-FS repository!***
+
+Executing the `add-replica` command should produce output reporting on the steps being performed,
+and only take a couple of moments to complete.
+
+*If no output is produced and the command seems to be hanging, make sure that port 80 on your Stratum 0 server
+is accessible via the IP address you are using!*
+
+#### Bypassing the Geo API license key
+
+If you prefer not to create a MaxMind account and Geo API license key for the sake of this tutorial,
+you can bypass the "`CVMFS_GEO_LICENSE_KEY not set`" error message
+produced by `cvmfs_server add-replica` by defining the `$CVMFS_GEO_DB_FILE` environment variable
+to `NONE` before running the command:
+
+```bash
+# only do this if you do not want to provide a Geo API key (not recommended!)
+export CVMFS_GEO_DB_FILE=NONE
```

Comment:
```suggestion
produced by `cvmfs_server add-replica` by setting the server variable `CVMFS_GEO_DB_FILE` 
to `NONE` before running the command:

```bash
# only do this if you do not want to provide a Geo API key (not recommended!)
echo 'CVMFS_GEO_DB_FILE=NONE' | sudo tee -a /etc/cvmfs/server.local
```